handlers/src/bonapp.rs

```diff
@@ -0,0 +1,191 @@
+use core::ops::DerefMut;
+
+use axum::{
+	extract::Path,
+	response::{IntoResponse, Response},
+	Json,
+};
+use http::StatusCode;
+use tracing::instrument;
+
+const CAFE_NAME_MAP: phf::Map<&str, u32> = phf::phf_map! {
+	"stav-hall" => 261,
+	"the-cage" => 262,
+	"kings-room" => 263,
+	"burton" => 35,
+	"ldc" => 36,
+	"sayles" => 24,
+	"weitz" => 458,
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum BonAppProxyError {
+	#[error("error while encoding query string")]
+	QueryStringEncoding(#[from] serde_urlencoded::ser::Error),
+
+	#[error("error while sending proxied request to bonapp")]
+	Request(reqwest::Error),
+
+	#[error("error while processing proxied response from bonapp ({0})")]
+	Response(reqwest::Error),
+
+	#[error("unknown cafe")]
+	UnknownCafe,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+enum ProxyRequestQueryParameters {
+	Cafe { cafe: String },
+	Menu { cafe: String },
+	ItemNutrition { item: String },
+}
+
+#[test]
+fn query_parameters_serialize() {
+	let query = ProxyRequestQueryParameters::Cafe {
+		cafe: "foo".to_string(),
+	};
+	assert_eq!(
+		serde_urlencoded::to_string(query).ok(),
+		Some("cafe=foo".to_string())
+	);
+}
+
+#[derive(Debug)]
+enum QueryType {
+	Cafe,
+	Menu,
+	ItemNutrition,
```

Comment:
The nutrition handler, route, and types might look something like this? The types may need more checking but I'm seeing queries load using the proxied_query that's set up. What are your thoughts on this?

I am able to see data load for these three urls. I was initially fighting with optional empty string or integer values being returned for a few fields but that is looking to be solved.

```
http://localhost:3000/api/food/item/1
http://localhost:3000/api/food/item/18759698
http://localhost:3000/api/food/item/18759699
```

```diff
diff --git a/handlers/src/bonapp.rs b/handlers/src/bonapp.rs
index d4ea954..a79c6ed 100644
--- a/handlers/src/bonapp.rs
+++ b/handlers/src/bonapp.rs
@@ -177,6 +177,13 @@ pub async fn cafe_menu_handler(
 	proxied_query(QueryType::Menu, &cafe_id).await
 }
 
+#[instrument(skip_all)]
+pub async fn nutrition_handler(
+	Path(item_id): Path<String>,
+) -> Result<Json<types::food::ItemNutritionResponse>, BonAppProxyError> {
+	proxied_query(QueryType::ItemNutrition, &item_id).await
+}
+
 impl IntoResponse for BonAppProxyError {
 	fn into_response(self) -> axum::response::Response {
 		let text = self.to_string();
diff --git a/routes/src/food.rs b/routes/src/food.rs
index 45cf53d..d23978d 100644
--- a/routes/src/food.rs
+++ b/routes/src/food.rs
@@ -1,6 +1,6 @@
 use axum::{routing::get, Router};
 use handlers::{
-	bonapp::{cafe_handler, cafe_menu_handler, named_cafe_handler, named_cafe_menu_handler},
+	bonapp::{cafe_handler, cafe_menu_handler, named_cafe_handler, named_cafe_menu_handler, nutrition_handler},
 	github::pause_menu_handler,
 };
 
@@ -9,6 +9,7 @@ pub fn router() -> Router {
 	Router::new()
 		.route("/menu/:cafe_id", get(cafe_menu_handler))
 		.route("/cafe/:cafe_id", get(cafe_handler))
+		.route("/item/:item_id", get(nutrition_handler))
 		.route("/named/cafe/:name", get(named_cafe_handler))
 		.route("/named/menu/:name", get(named_cafe_menu_handler))
 		.route("/named/menu/the-pause", get(pause_menu_handler))
diff --git a/types/src/food.rs b/types/src/food.rs
index 1e8d6fa..0b4b28d 100644
--- a/types/src/food.rs
+++ b/types/src/food.rs
@@ -250,7 +250,7 @@ pub struct Goitem {
 
 #[derive(Serialize, Deserialize)]
 pub struct Nutrition {
-	kcal: String,
+	kcal: Option<KCalEnum>,
 	well_being: Option<String>,
 	well_being_image: String,
 }
@@ -345,9 +345,38 @@ pub struct Item {
 	monotony: Monotony,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct ItemNutrition {
+    id: String,
+    label: String,
+    description: String,
+    zero_entree: String,
+    raw_cooked: String,
+    is_rotating: String,
+    cor_icon: CorIconUnion,
+    price: String, // should change other Price to be String and not enum
+    sizes: Vec<Size>,
+    nutrition_summary: Nutrition,
+    nutrition_details: ItemNutritionDetails,
+    ingredients: String,
+    ingredient_details: String,
+    nutrition_info: String,
+    special: i64,
+    rating: String,
+    connector: String,
+    options: OptionsUnion,
+    vendors: Vec<Option<serde_json::Value>>, // should change this to somewhat match OptionsUnion
+    monotony: Monotony,
+}
+
 #[derive(Serialize, Deserialize)]
 pub struct Monotony {}
 
+#[derive(Serialize, Deserialize)]
+pub struct ItemNutritionResponse {
+	items: HashMap<String, ItemNutrition>
+}
+
 #[derive(Serialize, Deserialize)]
 pub struct ItemNutritionDetails {
 	calories: Option<Calories>,
@@ -517,3 +546,10 @@ pub enum StationEnum {
 	#[serde(rename = "<strong>@tortilla</strong>")]
 	StrongTortillaStrong,
 }
+
+#[derive(Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum KCalEnum {
+	String(String),
+	Integer(i64),
+}
```

Reply:
Yep, exactly! That's how this all works. I went ahead and added your change.

I think we'll want to do a big sweep over all the types and prune off the things we don't use on the client, like I say. That'll give us the opportunity to also match up types that are actually shared, too.

Reply:
> I think we'll want to do a big sweep over all the types and prune off the things we don't use on the client

I see a few follow up commits addressing types -- are we tracking what we have validated?